src/streams/EditStreamCard.js

```diff
@@ -63,28 +63,25 @@ type PropsCreateStream = $ReadOnly<{|
 type Props = $ReadOnly<PropsEditStream | PropsCreateStream>;
 
 function explainCreateWebPublicStreamPolicy(
-  policy: CreateWebPublicStreamPolicyT,
+  policy: CreateWebPublicStreamPolicy,
   realmName: string,
 ): LocalizableText {
   return {
     text: (() => {
       switch (policy) {
-        // FlowIssue: sad that we end up having to write numeric literals here :-/
-        //   But the most important thing to get from the type-checker here is
-        //   that the ensureUnreachable works -- that ensures that when we add a
-        //   new possible value, we'll add a case for it here.  Couldn't find a
-        //   cleaner way to write this that still accomplished that. Discussion:
-        //     https://github.com/zulip/zulip-mobile/pull/5384#discussion_r875147220
-        case 6: // CreateWebPublicStreamPolicy.Nobody
+        case CreateWebPublicStreamPolicy.Nobody:
           return '{realmName} does not allow anybody to make web-public streams.';
-        case 7: // CreateWebPublicStreamPolicy.OwnerOnly
+        case CreateWebPublicStreamPolicy.OwnerOnly:
           return '{realmName} only allows organization owners to make web-public streams.';
-        case 2: // CreateWebPublicStreamPolicy.AdminOrAbove
+        case CreateWebPublicStreamPolicy.AdminOrAbove:
           return '{realmName} only allows organization administrators or owners to make web-public streams.';
-        case 4: // CreateWebPublicStreamPolicy.ModeratorOrAbove
+        case CreateWebPublicStreamPolicy.ModeratorOrAbove:
           return '{realmName} only allows organization moderators, administrators, or owners to make web-public streams.';
+        /* $FlowExpectedError[invalid-exhaustive-check]: TODO: Flow points
+           out that the default case can be removed because the other cases
+           are exhaustive. ESLint doesn't agree; fix that and remove the default
+           case. */
```

Comment:
```
flow: Use a Flow enum for the first time! (CreateWebPublicStreamPolicy)
```

Pasted from the commit message, here's what ESLint reports when I remove the default case:

```
  /Users/chrisbobbe/dev/zulip-mobile/src/streams/EditStreamCard.js
    70:15  error  Expected to return a value at   consistent-return
                  the end of arrow function
    71:7   error  Expected a default case         default-case
```

Thinking more about this, I think it's not realistic to expect `consistent-return` or `default-case` (or any ESLint rule) to recognize when the `case`s are exhaustive. We might teach ESLint to better understand Flow syntax, as we do in

```
eslint: Use new hermes-eslint parser, instead of @babel/eslint-parser
```

. But we can't expect it to actually do the type-checking necessary to conclude that the `case`s are exhaustive; ESLint isn't a typechecker.

So: can we repeal or adjust [`consistent-return`](https://eslint.org/docs/latest/rules/consistent-return) and [`default-case`](https://eslint.org/docs/latest/rules/default-case)?

Reply:
Yeah, I think the right answer here is to disable those two rules.

The one thing I see that we'll lose by doing so is that if you have a function whose return type includes `void`, then I'd like something like `consistent-return` to still apply: if there are any `return` statements with arguments, then I'd like to not have other control paths that implicitly return. That just makes it a bit easier when reading the code to be confident of spotting all the ways it returns.

But I don't think that's critical. The type-checker is taking care of everything we need here that's actually critical. We can cheerfully disable these rules.